broker-daemon/utils/migrate-store.spec.js

```diff
@@ -1,117 +1,192 @@
+  let resumeStub
 
   beforeEach(() => {
-    stream = {
-      on: sinon.stub()
-    }
+    onStub = sinon.stub()
+    pauseStub = sinon.stub()
+    resumeStub = sinon.stub()
     sourceStore = {
-      createReadStream: sinon.stub().returns(stream)
+      createReadStream: sinon.stub().returns({
+        on: onStub,
+        pause: pauseStub,
+        resume: resumeStub
+      })
     }
     targetStore = {
-      batch: sinon.stub().callsArgAsync(1)
+      batch: sinon.stub()
     }
-    fakeRecords = [
-      { key: '1', value: 'some value' },
-      { key: '2', value: 'another value' },
-      { key: '3', value: '3rd value' }
-    ]
-    fakeOperations = [
-      { type: 'put', key: 'a', value: '1' },
-      { type: 'put', key: 'b', value: '2' },
-      { type: 'put', key: 'c', value: '3' }
-    ]
-
-    stream.on.callsFake((evt, fn) => {
-      if (evt === 'error') return
-      const data = evt === 'data' ? fn : function () {}
-      const end = evt === 'end' ? fn : function () {}
-      const records = fakeRecords.slice()
-
-      const nextRecord = () => {
-        if (records.length) {
-          data(records.shift())
-          process.nextTick(nextRecord)
-        } else {
-          end()
-        }
-      }
-
-      process.nextTick(nextRecord)
-    })
     createDbOperation = sinon.stub()
-    fakeRecords.forEach(({ key, value }, i) => {
-      createDbOperation.withArgs(key, value).returns(fakeOperations[i])
-    })
+    batchSize = 1
   })
 
   it('returns a promise', () => {
-    expect(migrateStore(sourceStore, targetStore, createDbOperation)).to.be.a('promise')
+    expect(migrateStore(sourceStore, targetStore, createDbOperation, batchSize)).to.be.a('promise')
   })
 
-  it('creates a readstream from the store', () => {
-    migrateStore(sourceStore, targetStore, createDbOperation)
-
+  it('creates a read string from the source sublevel store', async () => {
+    migrateStore(sourceStore, targetStore, createDbOperation, batchSize)
+    expect(sourceStore.createReadStream).to.have.been.called()
     expect(sourceStore.createReadStream).to.have.been.calledOnce()
   })
 
-  it('sets an error handler', () => {
-    migrateStore(sourceStore, targetStore, createDbOperation)
-
-    expect(stream.on).to.have.been.calledWith('error', sinon.match.func)
+  context('on error', () => {
+    it('rejects the promise if there was a failure', () => {
+      const promise = migrateStore(sourceStore, targetStore, createDbOperation, batchSize)
+      const error = onStub.args[2][1]
+      error()
+      return expect(promise).to.eventually.be.rejected()
+    })
   })
 
-  it('sets an end handler', () => {
-    migrateStore(sourceStore, targetStore, createDbOperation)
+  context('on end', () => {
+    let promisifyStub
+    let batchStub
+    let reverts
 
-    expect(stream.on).to.have.been.calledWith('end', sinon.match.func)
-  })
+    beforeEach(() => {
+      batchStub = sinon.stub()
+      promisifyStub = sinon.stub().returns(batchStub)
```

Comment:
A `withArgs` here might be nice to be more specific about what the promisify is wrapping.